src/modules/Users.pm

```diff
@@ -5750,15 +5751,12 @@ sub ReadNISMaster {
 BEGIN { $TYPEINFO{ReadNISAvailable} = ["function", "boolean"];}
 sub ReadNISAvailable {
 
-    my $passwd_source = SCR->Read (".etc.nsswitch_conf.passwd");
-    if (defined $passwd_source) {
-	foreach my $source (split (/ /, $passwd_source)) {
+    my $passwd_source = Nsswitch->ReadDb ("passwd");
 
-	    if ($source eq "nis" || $source eq "compat") {
-		return (Package->Installed ("ypbind") && Service->Status ("ypbind") == 0);
-	    }
-	}
+    if ( grep ( /^nis|compat$/, @$passwd_source ) ) {
```

Comment:
I think this should be `/^(nis|compat)$/`. As it is now, I think these strings satisfies the regexp: "whatevercompat", "niswhatever". But it's regular expressions, so I could be of course wrong.

Reply:
No, you are not wrong. You're perfectly right. Thanks for catching it :)